src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,933 @@
+	 *                                specified store namespace.
+	 * @return array The configuration for the specified store namespace. This will be the updated configuration if a
+	 *               $config argument was provided.
+	 */
+	public function config( string $store_namespace, array $config = array() ): array {
+		if ( ! isset( $this->config_data[ $store_namespace ] ) ) {
+			$this->config_data[ $store_namespace ] = array();
+		}
+		if ( is_array( $config ) ) {
+			$this->config_data[ $store_namespace ] = array_replace_recursive(
+				$this->config_data[ $store_namespace ],
+				$config
+			);
+		}
+		return $this->config_data[ $store_namespace ];
+	}
+
+	/**
+	 * Prints the serialized client-side interactivity data.
+	 *
+	 * Encodes the config and initial state into JSON and prints them inside a
+	 * script tag of type "application/json". Once in the browser, the state will
+	 * be parsed and used to hydrate the client-side interactivity stores and the
+	 * configuration will be available using a `getConfig` utility.
+	 *
+	 * @since 6.5.0
+	 */
+	public function print_client_interactivity_data() {
+		$store      = array();
+		$has_state  = ! empty( $this->state_data );
+		$has_config = ! empty( $this->config_data );
+
+		if ( $has_state || $has_config ) {
+			if ( $has_config ) {
+				$store['config'] = $this->config_data;
+			}
+			if ( $has_state ) {
+				$store['state'] = $this->state_data;
+			}
+			wp_print_inline_script_tag(
+				wp_json_encode(
+					$store,
+					JSON_HEX_TAG | JSON_HEX_AMP
+				),
+				array(
+					'type' => 'application/json',
+					'id'   => 'wp-interactivity-data',
+				)
+			);
+		}
+	}
+
+	/**
+	 * Registers the `@wordpress/interactivity` script modules.
+	 *
+	 * @since 6.5.0
+	 */
+	public function register_script_modules() {
+		$suffix = wp_scripts_get_suffix();
+
+		wp_register_script_module(
+			'@wordpress/interactivity',
+			includes_url( "js/dist/interactivity$suffix.js" )
+		);
+
+		wp_register_script_module(
+			'@wordpress/interactivity-router',
+			includes_url( "js/dist/interactivity-router$suffix.js" ),
+			array( '@wordpress/interactivity' )
+		);
+	}
+
+	/**
+	 * Adds the necessary hooks for the Interactivity API.
+	 *
+	 * @since 6.5.0
+	 */
+	public function add_hooks() {
+		add_action( 'wp_enqueue_scripts', array( $this, 'register_script_modules' ) );
+		add_action( 'wp_footer', array( $this, 'print_client_interactivity_data' ) );
+	}
+
+	/**
+	 * Processes the interactivity directives contained within the HTML content
+	 * and updates the markup accordingly.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @uses WP_Interactivity_API::data_wp_interactive_processor()
+	 * @uses WP_Interactivity_API::data_wp_context_processor()
+	 * @uses WP_Interactivity_API::data_wp_bind_processor()
+	 * @uses WP_Interactivity_API::data_wp_class_processor()
+	 * @uses WP_Interactivity_API::data_wp_style_processor()
+	 * @uses WP_Interactivity_API::data_wp_text_processor()
+	 *
```

Comment:
Adding additional references for static analysis to avoid complaints about unused methods:

```suggestion
	 * @uses WP_Interactivity_API::data_wp_router_region_processor()
	 * @uses WP_Interactivity_API::data_wp_each_processor()
	 *
```

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/5953/commits/8eec2f3c42958482f459153b862172d6b8fa9130.